### DISCUSSION TITLE
Adding the ability to work with custom dictionaries

### DISCUSSION THREAD

**Discussion by uplqu** [ardha27/AI-Waifu-Vtuber#41] [Ideas]
Is it possible to connect custom dictionaries to automatically convert the output text (for use in TTS)? This is needed for a more correct pronunciation by the language model of words and numbers (e.g. SileroTTS according to the rules of its [SSML](https://github.com/snakers4/silero-models/wiki/SSML))
I can provide such .dic and .rex files if needed.

**Comment by uplqu** [comment]
The structure is like this

- Add user dictionary of the .dic format
- Read file output.txt
- Convert text according to dictionary rules
- Save the corrected text in a file
- TTS model reads text from the file

Is it possible to make this work in this case?

**Comment by ardha27** [comment]
Actually it's my first time heard about dic and rex files. But from what you saying, i think it's possible to add it

**Reply by uplqu** [reply]
In essence, these files are just a set of rules for what words or numbers to replace with what.
It can be opened with any text editor. 

And the rules inside such files have this structure:
10=ten
11=eleven

Here is one of these files for example.
[01_numbers-en.zip](https://github.com/ardha27/AI-Waifu-Vtuber/files/11244764/01_numbers-en.zip)


It would be great to see such a feature here in the future.
